Modules/xASL_module_ASL.m

```diff
@@ -874,3 +512,305 @@
+        fprintf('%s\n', ['Detected the following unique ' parNames{iPar} 's (ms):'])
+        for ivol = 1:x.Q.(['Number' parNames{iPar}])
+            fprintf('%.2f, ', round(x.Q.(['unique' parNames{iPar}])(iVol), 4));
+        end
+        fprintf('\n');
+    end
+    
+    % If multi-parameter data is detected, we switch on multi-parameter quantification by default. Unless this has been deactived in dataPar.json
+    if x.Q.(['Number' parNames{iPar}])>1
+	    if ~isfield(x.modules.asl, ['bQuantifyMulti' parAbbreviation{iPar}]) || isempty(x.modules.asl.(['bQuantifyMulti' parAbbreviation{iPar}]))
+		    % By default, initialize as Multi-par quantification as true
+		    x.modules.asl.(['bQuantifyMulti' parAbbreviation{iPar}]) = true;
+	    end
+    else
+	    x.modules.asl.(['bQuantifyMulti' parAbbreviation{iPar}]) = false;
+    end
+    
+    % Deal with too short vectors (e.g., repetitions in the case of single-parameter)
+    nPar = length(x.Q.([parNames{iPar}]));
+    if mod(nVolumes, nPar)~=0
+        error(['Number of ' parNames{iPar} 's (n=' xASL_num2str(nPar) ') does not fit in nVolumes (n=' xASL_num2str(nVolumes) ')']);
+        % If we don't issue an error here, repmat will crash
+    end
+    
+    factorPar = nVolumes/nPar;
+    x.Q.(parNames{iPar}) = repmat(x.Q.(parNames{iPar})(:), [factorPar 1]);
+    nPar = length(x.Q.(parNames{iPar}));
+    
+    % Number of echoes should now be equal to the number of ASL volumes
+    if nPar~=nVolumes
+        warning(['Number of ' parNames{iPar} ' (n=' xASL_num2str(nPar) ') should be equal to number of ASL volumes (n=' xASL_num2str(nVolumes) ')']);
+    end
+
+end
+
+
+
+%% 4. TimeEncoded parsing
+% Check if TimeEncoded is defined
+if isfield(x.Q, 'TimeEncodedMatrixType') || isfield(x.Q, 'TimeEncodedMatrixSize') || isfield(x.Q, 'TimeEncodedMatrix')
+    fprintf(2,'Time encoded data detected, we will process this but this is a new feature that is still under development\n');
+    x.modules.asl.bTimeEncoded = true;
+	
+	% Initialize as empty if missing
+	if ~isfield(x.Q, 'TimeEncodedMatrixType')
+		x.Q.TimeEncodedMatrixType = [];
+	end
+
+	if ~isfield(x.Q, 'TimeEncodedMatrixSize')
+		x.Q.TimeEncodedMatrixSize = [];
+	end
+
+	if ~isfield(x.Q, 'TimeEncodedMatrix')
+		x.Q.TimeEncodedMatrix = [];
+	end
+
+	% Check for duplicate definitions
+	if isempty(x.Q.TimeEncodedMatrixType) && isempty(x.Q.TimeEncodedMatrix)
+		error('Neither TimeEncodedMatrixType nor TimeEncodedMatrix were provided for a TimeEncoded acquisition.')
+	end
+	
+	if isempty(x.Q.TimeEncodedMatrixSize)
+		if isempty(x.Q.TimeEncodedMatrix)
+			error('TimeEncodedMatrixSize not defined and TimeEncodedMatrix not provided.')
+		else
+			x.Q.TimeEncodedMatrixSize = size(x.Q.TimeEncodedMatrix,1);
+		end
+	else
+		if (x.Q.TimeEncodedMatrixSize < 2)
+			fprintf(2,'TimeEncodedMatrixSize field missing (should be a multiple of 4)...\n');
+		end
+		if (x.Q.TimeEncodedMatrixSize ~= size(x.Q.TimeEncodedMatrix, 1))
```

Comment:
Please change this line to. We did that in 1584, but best that we do it here as well to avoid loosing that in the conflict resolving.

if ~isempty(x.Q.TimeEncodedMatrix) && (x.Q.TimeEncodedMatrixSize ~= size(x.Q.TimeEncodedMatrix, 1))

Reply:
Fixed in